test/org/javarosa/xform/parse/XFormParserTest.java

```diff
@@ -202,6 +202,10 @@ private void timeParsing(LargeInstanceFileGenerator lfg, Path largeDataFilename,
         serAndDeserializeForm(r("Simpler_Cascading_Select_Form.xml"));
     }
 
+    @Test public void allWidgetsFormSavesAndRestores() throws IOException, DeserializationException {
```

Comment:
I'm concerned that people reviewing this won't understand why this test is here and why it's important. Using all-widgets is a good idea, but here we need to make sure one can understand that this is a test that will prevent regression if the RangeWidget somehow disappears from the list of classes for JR serialization.

I'd suggest using a new form with just a range widget and adding a comment inside the test linking to the GitHub issue.

Reply:
already done but we lose coverage on serialize/deserialize `OSMTag `, `GroupDef`,  and `SelectChoice` though. I squashed the commit, i assume this is okay?
Thanks for the review